floodmodeller_api/test/test_toolbox_structure_log.py

```diff
@@ -79,156 +83,110 @@ def structure():
+    expected = """Unit Name,Unit Type,Unit Subtype,Comment,Friction,Dimensions (m),Weir Coefficient,Culvert Inlet/Outlet Loss
+first,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00 (Total conduit length: 40.00),,
+second,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+third,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+fourth,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+fifth,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 0.00,,
+"""
+
+    slb._dat = conduit_chain_dat
+    tmp_csv = Path(tmpdir) / "test_multi_conduits.csv"
+    print(tmp_csv)
+    with tmp_csv.open("w", newline="") as file:
         slb._writer = csv.writer(file)
         slb._add_conduits()
+        slb._write_csv_output(file)
 
+    with open(tmp_csv) as read_file:
+        text = read_file.read()
 
-def test_add_structures(slb, structure, tmpdir):
-    slb._dat = DAT()
-    structure.soffit = 3
-    structure.weir_coefficient = 1
-    structure.data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0]})
-    structure.section_data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0], "Mannings n": [0, 0]})
-    structure.opening_data = pd.DataFrame(
-        data={"Start": 0, "Finish": 0, "Springing Level": 0, "Soffit Level": 0},
-        index=[0],
-    )
-    structure.crest_elevation = 1
-    structure.weir_breadth = 1
-    structure.weir_length = 1
-    structure.weir_elevation = 1
-    slb._dat.structures["test_structure_orifice"] = structure
-    struc_spill = copy.deepcopy(structure)
-    struc_spill._unit = "SPILL"
-    slb._dat.structures["test_structure_spill"] = struc_spill
-    struc_sluice = copy.deepcopy(structure)
-    struc_sluice._unit = "SLUICE"
-    slb._dat.structures["test_structure_sluice"] = struc_sluice
-    struc_rnweir = copy.deepcopy(structure)
-    struc_rnweir._unit = "RNWEIR"
-    slb._dat.structures["test_structure_rnweir"] = struc_rnweir
-    struc_weir = copy.deepcopy(structure)
-    struc_weir._unit = "WEIR"
-    slb._dat.structures["test_structure_weir"] = struc_weir
-    struc_bridge = copy.deepcopy(structure)
-    struc_bridge._unit = "BRIDGE"
-    slb._dat.structures["test_structure_bridge"] = struc_bridge
-    struc_none = copy.deepcopy(structure)
-    struc_none._unit = "NONE"
-    slb._dat.structures["test_structure_none"] = struc_none
-
-    tmp_csv = Path(tmpdir) / "temp_structure_data.csv"
-    with tmp_csv.open("w") as file:
-        slb._writer = csv.writer(file)
-        slb._add_structures()
+    assert text == expected
+
+
+def test_full_dat_from_python(slb, tmpdir, ex18_dat_path, ex18_dat_expected):
+    # these two tests should be as described in the toolbox documentation
+
+    # Im not sure if this is slightly redundant compared to test from commandline
```

Comment:
I understand what you mean! I think it's worth keeping it though as it might help with diagnosing problems if ever something got broken e.g. if `test_full_dat_from_commandline` was failing but this one wasn't, then we'd immediately know the problem was with the command line part.